compiler/qsc_hir/src/ty.rs

```diff
@@ -220,11 +238,26 @@ impl Display for GenericParam {
 #[derive(Clone, Debug, PartialEq)]
 pub enum GenericParam {
     /// A type parameter.
-    Ty,
+    Ty(TypeParamName),
```

Comment:
I'm concerned that giving them NodeIds and calling them Idents implies that they need to be mappable in the type map.

Reply:
Hmm, that's a concern I hadn't thought of. But generic params are already represented by `Ident`s in the AST, and the `NodeId`s used for the types table are for the _AST_ types. The types table already references generic parameters by their `NodeId`. So I think that's all consistent:

https://github.com/microsoft/qsharp/blob/81873fe2709f05d8b7b9671b53dd72a79d3d6596/compiler/qsc_frontend/src/typeck.rs#L28-L30

Similarly, the names table in the resolver also uses AST `NodeId`s. Except maybe the comment may be incorrect, since all `Ident` nodes may not get mapped? I'm guessing generic parameter `Ident`s are excluded?

https://github.com/microsoft/qsharp/blob/81873fe2709f05d8b7b9671b53dd72a79d3d6596/compiler/qsc_frontend/src/resolve.rs#L31-L33

Reply:
From poking around a bit, I found that the ids of the AST Idents are mappable in the names map, but not in the type map, which is fine and consistent with the comments describing those mappings.

Reply:
Another issue I'm running into is that `Ty`'s don't have Idents on them, and adding an Ident to the ty.rs would ruin the separation between the hir.rs and the ty.rs modules.

Reply:
But `ty.rs` already imports a ton of stuff from `hir.rs`. Either way I'm okay with this - having a more specific type is not such a bad thing after all.